types/geojson/index.d.ts

```diff
@@ -1,123 +1,157 @@
-// Type definitions for GeoJSON Format Specification Revision 1.0
-// Project: http://geojson.org/
+// Type definitions for geojson 7946.0
+// Project: https://geojson.org/
 // Definitions by: Jacob Bruun <https://github.com/cobster>
 //                 Arne Schubert <https://github.com/atd-schubert>
+//                 Jeff Jacobson <https://github.com/JeffJacobson>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
+// Note: as of the RFC 7946 version of GeoJSON, Coordinate Reference Systems
+// are no longer supported. (See https://tools.ietf.org/html/rfc7946#appendix-B)}
+
 export as namespace GeoJSON;
 
-/***
- * http://geojson.org/geojson-spec.html#geojson-objects
+/**
+ * The valid values for the "type" property of GeoJSON geometry objects.
+ * https://tools.ietf.org/html/rfc7946#section-1.4
  */
-export interface GeoJsonObject {
-    type: string;
-    bbox?: number[];
-    crs?: CoordinateReferenceSystem;
-}
+export type GeoJsonGeometryTypeNames = "Point" | "LineString" | "MultiPoint" | "Polygon" | "MultiLineString" |
+    "MultiPolygon" | "GeometryCollection";
+
+/**
+ * The value values for the "type" property of GeoJSON Objects.
+ * https://tools.ietf.org/html/rfc7946#section-1.4
+ */
+export type GeoJsonTypeNames = "FeatureCollection" | "Feature" | GeoJsonGeometryTypeNames;
+
+/**
+ * Bounding box
+ * https://tools.ietf.org/html/rfc7946#section-5
+ */
+export type BoundingBox = [number, number, number, number] | [number, number, number, number, number];
 
-/***
- * http://geojson.org/geojson-spec.html#positions
+/**
+ * A Position is an array of coordinates.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.1
  */
-export type Position = number[];
+export type Position = [number, number] | [number, number, number];
 
-/***
- * http://geojson.org/geojson-spec.html#geometry-objects
+/**
+ * The base GeoJSON object.
+ *  https://tools.ietf.org/html/rfc7946#section-3
  */
-export interface DirectGeometryObject extends GeoJsonObject {
-    coordinates: Position[][][] | Position[][] | Position[] | Position;
+export interface GeoJsonObject {
+    /**
+     * Allow foreign members as specified in https://tools.ietf.org/html/rfc7946#section-6.1
+     */
+    [key: string]: any;
```

Comment:
Doing this here pretty much destroys type safety. The compiler will assume any unknown (or possibly misspelled) property name has the `any` type instead of flagging it as an error. This is especially dangerous now that later versions of TypeScript don't require subscripts for indexed properties:

```ts
let x: GeoJsonObject;
console.log(x.typo); // Intentional property access, or typo of `type'?
```

I suggest omitting this and recommending that people who need this feature either extend the relevant interface or use an `&` type.

Reply:
Agreed that this will destroy any type safety.